lib/validation.js

```diff
@@ -22,12 +22,34 @@ function compileSchemasForSerialization (context, compile) {
         throw new Error('response schemas should be nested under a valid status code, e.g { 2xx: { type: "object" } }')
       }
 
-      acc[statusCode] = compile({
-        schema,
-        url,
-        method,
-        httpStatus: statusCode
-      })
+      // ducktyping to check if the media content is meant to be used or it's the short form.
+      let mediaContent = false
+      if (schema.content) {
+        const firstEntry = schema.content[Object.keys(schema.content)[0]]
+        mediaContent = !!(typeof firstEntry === 'object' && firstEntry.schema)
```

Comment:
I'm checking here to know whether we have different content types or not, to set the `acc[statusCode]` accurately
```JS
      if (mediaContent) {
        const contentTypesSchemas = {}
        for (const mediaName in schema.content) {
          const contentSchema = schema.content[mediaName].schema
          contentTypesSchemas[mediaName] = compile({
            schema: contentSchema,
            url,
            method,
            httpStatus: statusCode
          })
        }
        acc[statusCode] = contentTypesSchemas
      } else {
        acc[statusCode] = compile({
          schema,
          url,
          method,
          httpStatus: statusCode
        })
      }
```

so the output will be  for example: 

```JS
{
  '200': { // in case of different content types
    'application/json': [Function: main],
    'application/vnd.v2+json': [Function: main],
    'application/vnd.v1+json': [Function: main]
  }, 
  '202': [Function: main] // else 
}
```

kindly, can you elaborate more on how this can be achieved without that check and by passing `contentType` to the compile function?

Reply:
> can you elaborate more on how this can be achieved without that check

You can normalize the `schema.content`
When `schema.content` is set, the `normalizeSchema` function should assure that you have multiple content types, otherwise, run the `else` statement.

> and by passing contentType to the compile function?

It is not related to the initial check.
Users may want to customize how they compile the schema based on that argument, it is just a good addition.

Reply:
Ahh ok, got it, thank you!
Going to commit the changes.